geomstats/geometry/hpd_matrices.py

```diff
@@ -510,7 +510,7 @@ class HPDAffineMetric(ComplexRiemannianMetric):
     """
 
     def __init__(self, n, power_affine=1, scale=1.0):
-        dim = int(n * (n + 1) / 2)
+        dim = n**2
```

Comment:
it was correct. the behavior you want is achieved by making `default_coords_type="intrisinc"` (it was wrongly defined as `"intrinsic"`.

Please revert this change here and in the other places.

For future reference, it is also advised to make corrections for code that is not the focus of the PR in different PRs.

Reply:
Hello @lpereira95, 
I am not sure about the HPD matrices manifold dimension.
I would say that its dimension as a real scalar coordinates is `n**2`: 
- the dimension of the real part is equal to `n * (n+1) / 2` as it entirely determine by the upper triangular part, diagonal included.
- the dimension of the imaginary part is equal to `n * (n-1) / 2` as it entirely determine by the upper triangular part, diagonal excluded.

We can also say that the dimension is equal the `n * (n+1) / 2` as it entirely determine by the upper triangular part (diagonal included) which is composed of `n * (n+1) / 2` complex coefficients.
However, we can not create a complex basis of this dimension.
For example the diagonal coefficients of an HPD matrix are positive reals. If a diagonal coefficient is multiplied by a complex constant, then the matrix is not HPD anymore. 
Anyway, the HPD matrices is not a vector space as multiplying an HPD matrix by minus 1 do not give an HPD matrix.

Even if the HPD matrices manifold is not a vector space, if we had to define a basis of this space, I think that it would be better to use `n**2` basis vectors (real or complex) and using real scalar coordinates than to use `n * (n+1) / 2` complex basis vectors with complex coordinates.
Then I would suggest to consider that the dimension of the HPD matrices is rather `n**2` than `n * (n+1) / 2`.

So from a mathematical point of view, I would say that its dimension is `n**2` : an HPD matrix can be represented by `n**2` real coordinates.

However, if you consider the dimension as the minimum number of coefficients of the matrix that you need to save to be able to recover the entire HPD matrix, then it is `n * (n+1) / 2` complex coefficients.

What do you think about it?

Reply:
Hello @lpereira95  and @johnharveymath,
Whether you decide that the dimension is `n**2` or `n * (n+1) / 2`, you will have odd dimensions for some values of `n` in both cases. (For example `n=1`.)
If do not think that the HPD matrices manifold should be considered as a complex manifold.
For example, in dimension `n=1`, the HPD manifold is just the set of positive reals.
Then, taking the basis vector `[[1]]`, each one-dimensional HPD matrix `[[a]] = a * [[1]] `can be represented in this basis by the real coordinate `a`.
It is then an open subset of the real space of dimension 1 as the set of HPD matrices corresponds to the coordinate `a > 0`.
However, if we consider complex coordinates `a`, then `a * [[1]]` is not an HPD matrix in general.
The HPD matrices manifold is a manifold which has some complex values, but it is much closer to an open subset of a real vector space (with some real or complex basis vectors) than to a complex vector space.

In dimension `n=2`, the basis that I would choose would be the 4 basis vectors:
- `e1 = [[1, 0], [0, 0]]`
- `e2 = [[0, 0], [0, 1]]`
- `e3 = [[0, 1], [1, 0]]`
- `e4 = [[0, i], [-i, 0]]`

Then each two-dimensional HPD matrix can be represented as a linear combination of these 4 basis vectors using real coordinates: `hpd_matrix = a1 e1 + a2 e2 + a3 e3 + a4 e4`. 
The set of coefficients `(a1, a2, a3, a4) in R^4` corresponding to HPD matrices is an open convex subset of `R^4` which is stable by multiplication by a strictly positive constant.

Then would you agree to consider that its dimension is `n**2`?

Reply:
I see - it is embedded in a complex space but is not itself complex. Then `n**2` looks right to me, yes.